plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -86,6 +86,189 @@ protected String getStorageMetadataLocation(String materializedViewName)
+        Session incrRefreshEnabled = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", "incremental_refresh_enabled", "true")
+                .build();
+
+        String matViewDef = "SELECT a, b FROM source_table WHERE a < 3 OR a > 5";
+
+        // create source table and two identical MVs
+        assertUpdate("CREATE TABLE source_table (a int, b varchar)");
+        assertUpdate("INSERT INTO source_table VALUES (1, 'abc'), (2, 'def')", 2);
+        assertUpdate("CREATE MATERIALIZED VIEW mat_view_test_1 AS %s".formatted(matViewDef));
+        assertUpdate("CREATE MATERIALIZED VIEW mat_view_test_2 AS %s".formatted(matViewDef));
+
+        // execute first refresh: afterwards both MVs will contain: (1, 'abc'), (2, 'def')
+        assertUpdate("REFRESH MATERIALIZED VIEW mat_view_test_1", 2);
+        assertUpdate("REFRESH MATERIALIZED VIEW mat_view_test_2", 2);
+
+        // add some new rows to source
+        assertUpdate("INSERT INTO source_table VALUES (3, 'ghi'), (4, 'jkl'), (5, 'mno'), (6, 'pqr')", 4);
+
+        // will do incremental refresh, and only add: (6, 'pqr')
+        assertUpdate(incrRefreshEnabled, "REFRESH MATERIALIZED VIEW mat_view_test_1", 1);
+        // will do full refresh, and (re)add: (1, 'abc'), (2, 'def'), (6, 'pqr')
+        assertUpdate(defaultSession, "REFRESH MATERIALIZED VIEW mat_view_test_2", 3);
+
+        // cleanup
+        assertUpdate("DROP MATERIALIZED VIEW mat_view_test_1");
+        assertUpdate("DROP MATERIALIZED VIEW mat_view_test_2");
+        assertUpdate("DROP TABLE source_table");
+    }
+
+    @Test
+    public void testIncrementalRefreshDisabledForAggregation()
+    {
+        Session incrRefreshEnabled = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", "incremental_refresh_enabled", "true")
+                .build();
+
+        String matViewDef = "SELECT b, sum(a) AS total FROM source_table GROUP BY b";
+
+        // create source table and MV
+        assertUpdate("CREATE TABLE source_table (a int, b varchar)");
+        assertUpdate("INSERT INTO source_table VALUES (1, 'abc'), (2, 'def')", 2);
+        assertUpdate("CREATE MATERIALIZED VIEW mat_view_test_1 AS %s".formatted(matViewDef));
+
+        // execute first refresh: afterwards MV will contain: ('abc', 1), ('def', 2)
+        assertUpdate(incrRefreshEnabled, "REFRESH MATERIALIZED VIEW mat_view_test_1", 2);
+
+        // add some new rows to source
+        assertUpdate("INSERT INTO source_table VALUES (3, 'ghi'), (4, 'jkl'), (5, 'abc'), (6, 'ghi')", 4);
+
+        // will perform full refresh: ('abc', 6), ('def', 2), ('ghi', 9), ('jkl', 4)
+        assertUpdate(incrRefreshEnabled, "REFRESH MATERIALIZED VIEW mat_view_test_1", 4);
+
+        // cleanup
+        assertUpdate("DROP MATERIALIZED VIEW mat_view_test_1");
+        assertUpdate("DROP TABLE source_table");
+    }
+
+    @Test
+    public void testIncrementalRefreshDisabledForJoin()
```

Comment:
most of these, `aggregation/join/subquery/distinct` should really be part of engine tests suite. We could have just one `full` refresh test here